### PR TITLE
Add page webhooks

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to add page webhooks `PAGE_CREATED`, `PAGE_UPDATED` and `PAGE_DELETED`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by d-wysocki** [comment]
Code is ready for review, now I'm working on documentation.

File: saleor/graphql/page/mutations/pages.py [hunk 1]
```diff
@@ -108,6 +108,11 @@ def _save_m2m(cls, info, instance, cleaned_data):
         if attributes:
             AttributeAssignmentMixin.save(instance, attributes)
 
+    @classmethod
+    def save(cls, info, instance, cleaned_input):
+        super(PageCreate, cls).save(info, instance, cleaned_input)
```
Comment (IKarbowiak) [inline saleor/graphql/page/mutations/pages.py:113]: I think `super().save(info, instance, cleaned_input)` is enough.

File: saleor/graphql/page/mutations/pages.py [hunk 2]
```diff
@@ -123,6 +128,11 @@ class Meta:
         error_type_class = PageError
         error_type_field = "page_errors"
 
+    @classmethod
+    def save(cls, info, instance, cleaned_input):
+        super(PageCreate, cls).save(info, instance, cleaned_input)
```
Comment (IKarbowiak) [inline saleor/graphql/page/mutations/pages.py:133]: Should we run `info.context.plugins.page_created(instance)` when `Page` is updated?

File: saleor/graphql/page/mutations/pages.py [hunk 3]
```diff
@@ -135,6 +145,13 @@ class Meta:
         error_type_class = PageError
         error_type_field = "page_errors"
 
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        page = cls.get_instance(info, **data)
+        response = super().perform_mutation(_root, info, **data)
+        info.context.plugins.page_updated(page)
```
Comment (IKarbowiak) [inline saleor/graphql/page/mutations/pages.py:152]: Shouldn't `page_deleted` be called?

File: saleor/graphql/page/mutations/pages.py (hunk 2 above)
Comment (IKarbowiak) [reply saleor/graphql/page/mutations/pages.py:133]: As you call `save` method from `PageCreate`, the `page_created` method is called too.

File: saleor/graphql/page/mutations/pages.py (hunk 3 above)
Comment (d-wysocki) [reply saleor/graphql/page/mutations/pages.py:152]: Should be, good catch. thanks.

File: saleor/graphql/page/mutations/pages.py [hunk 4]
```diff
@@ -135,6 +145,13 @@ class Meta:
         error_type_class = PageError
         error_type_field = "page_errors"
 
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        page = cls.get_instance(info, **data)
+        response = super().perform_mutation(_root, info, **data)
+        info.context.plugins.page_deleted(page)
```
Comment (korycins) [inline saleor/graphql/page/mutations/pages.py:152]: Did you test it? AFAIK PK will be `None` here

File: saleor/plugins/webhook/tests/test_webhook.py [hunk 5]
```diff
@@ -209,6 +210,42 @@ def test_checkout_updated(mocked_webhook_trigger, settings, checkout_with_items)
...
+    mocked_webhook_trigger.assert_called_once_with(
+        WebhookEventType.PAGE_UPDATED, expected_data
+    )
+
+
+@mock.patch("saleor.plugins.webhook.plugin.trigger_webhooks_for_event.delay")
+def test_page_deleted(mocked_webhook_trigger, settings, page):
+    settings.PLUGINS = ["saleor.plugins.webhook.plugin.WebhookPlugin"]
+    manager = get_plugins_manager()
+    manager.page_deleted(page)
```
Comment (korycins) [inline saleor/plugins/webhook/tests/test_webhook.py:241]: To proper test it I would simulate the situation that page is deleted. You can do the similar steps like `ModelDeleteMutation`
```
db_id = instance.id
instance.delete()

instance.id = db_id
```

**Review by korycins — APPROVED** [review]

File: saleor/graphql/page/mutations/pages.py (hunk 2 above)
Comment (IKarbowiak) [reply saleor/graphql/page/mutations/pages.py:133]: What about that?

**Review by maarcingebala — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/page/tests/test_page.py [hunk 6]
```diff
@@ -956,25 +997,27 @@ def test_create_page_with_product_reference_attribute_required_no_references_giv
...
-                    title
-                    id
-                }
-                pageErrors {
-                    field
-                    code
-                    message
-                }
-              }
+PAGE_DELETE_MUTATION = """
```
Comment (fowczarek) [inline saleor/graphql/page/tests/test_page.py:1000]: Could you reformat this mutation?
It's easier to read when mutations and queries have consistent formatting.

File: saleor/graphql/page/tests/test_page.py [hunk 7]
```diff
@@ -264,6 +266,45 @@ def test_page_create_mutation(staff_api_client, permission_manage_pages, page_ty
...
+    content = get_graphql_content(response)
+    data = content["data"]["pageCreate"]
+    assert data["pageErrors"] == []
+    assert data["page"]["title"] == page_title
+    assert data["page"]["content"] == page_content
+    assert data["page"]["slug"] == page_slug
+    assert data["page"]["isPublished"] == page_is_published
+    assert data["page"]["pageType"]["id"] == page_type_id
+
+    assert mocked_webhook_trigger.called
```
Comment (fowczarek) [inline saleor/graphql/page/tests/test_page.py:305]: Why we don't check what data are passed to webhook(`assert_called_once_with`)?

File: saleor/graphql/page/tests/test_page.py [hunk 8]
```diff
@@ -983,6 +1026,23 @@ def test_page_delete_mutation(staff_api_client, page, permission_manage_pages):
...
+    variables = {"id": graphene.Node.to_global_id("Page", page.id)}
+    response = staff_api_client.post_graphql(
+        PAGE_DELETE_MUTATION, variables, permissions=[permission_manage_pages]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["pageDelete"]
+    assert data["page"]["title"] == page.title
+    with pytest.raises(page._meta.model.DoesNotExist):
+        page.refresh_from_db()
+    assert mocked_webhook_trigger.called
```
Comment (fowczarek) [inline saleor/graphql/page/tests/test_page.py:1043]: Why we don't check what data are passed to webhook(`assert_called_once_with`)?

File: saleor/graphql/page/tests/test_page.py [hunk 9]
```diff
@@ -1089,6 +1149,43 @@ def test_update_page(staff_api_client, permission_manage_pages, page):
...
+    )
+
+    # then
+    content = get_graphql_content(response)
+    data = content["data"]["pageUpdate"]
+
+    assert not data["pageErrors"]
+    assert data["page"]["title"] == page_title
+    assert data["page"]["slug"] == new_slug
+    assert mocked_webhook_trigger.called
```
Comment (fowczarek) [inline saleor/graphql/page/tests/test_page.py:1186]: Why we don't check what data are passed to webhook(`assert_called_once_with`)?

File: saleor/webhook/payloads.py [hunk 10]
```diff
@@ -354,3 +364,21 @@ def generate_sample_payload(event_name: str) -> Optional[dict]:
     else:
         payload = _generate_sample_order_payload(event_name)
     return json.loads(payload) if payload else None
+
+
+def generate_page_payload(page: Page):
```
Comment (fowczarek) [inline saleor/webhook/payloads.py:369]: All payload generators are above. Why this one is in this position?

**Review by fowczarek — APPROVED** [review]

**Pull request merged**